pytorch_lightning/trainer/trainer.py

```diff
@@ -1395,4 +1395,8 @@ def _on_exception(self):
             return
         # save a checkpoint for fault tolerant training. we don't use `log_dir` to minimize the chances of failure.
         file_path = os.path.join(self.default_root_dir, ".pl_auto_save.ckpt")
+
+        # if a previous `pl_auto_save` was saved, delete it.
+        if os.path.exists(file_path):
+            os.remove(file_path)
```

Comment:
this should use fsspec / cloud_io in case default_root_dir is not a local storage path

Reply:
@tchaton do we need this at all? at this point we know we will save a checkpoint so if a file exists we would overwrite it.

Reply:
For the double failure test, it didn't seem to overwrite the file. Might be worth to double check.